src/transformers/modeling_utils.py

```diff
@@ -1622,7 +1623,9 @@ def _check_and_enable_flash_attn_2(
         return config
 
     @classmethod
-    def _check_and_enable_sdpa(cls, config, hard_check_only: bool = False) -> PretrainedConfig:
+    def _check_and_enable_sdpa(
```

Comment:
Can you update as well
```
./src/transformers/models/idefics/modeling_idefics.py:954:    # Adapted from transformers.modeling_utils.PreTrainedModel._check_and_enable_sdpa
./src/transformers/models/idefics/modeling_idefics.py:956:    def _check_and_enable_sdpa(cls, config, hard_check_only: bool = False) -> PretrainedConfig:
./src/transformers/models/falcon/modeling_falcon.py:954:    # Adapted from transformers.modeling_utils.PreTrainedModel._check_and_enable_sdpa
./src/transformers/models/falcon/modeling_falcon.py:956:    def _check_and_enable_sdpa(cls, config, hard_check_only: bool = False) -> "PretrainedConfig":
```

Reply:
If this function is in multiple files, I guess it makes sense to handle the logic in `_autoset_attn_implementation`

Reply:
You changed the signature hence my suggestion